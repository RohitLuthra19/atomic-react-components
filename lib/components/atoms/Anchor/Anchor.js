// @flow
import * as React from 'react';
import styled from 'styled-components';
import Link from 'next/link';

import { buildUrl } from '../../../core/utils/url';
import styles from './Anchor.style';

type Props = {
  to: string | any,
  as?: string,
  children: Node,
  noLink?: boolean,
  className: '',
  inheritedStyles?: string,
  handleLinkClick: (SyntheticEvent<>) => void,
  shallow?: boolean,
  scroll?: boolean,
  forceClick?: boolean,
};

const Anchor = ({
  children,
  to,
  as,
  className,
  noLink,
  inheritedStyles,
  handleLinkClick,
  shallow,
  scroll,
  forceClick,
  ...others
}: Props): Node => (noLink ? (
  <a href={buildUrl(to)} className={className} onClick={handleLinkClick} {...others}>
    {children}
  </a>
) : (
  <Link href={to} as={as} shallow={shallow} scroll={scroll}>
    {forceClick
      ? (
        <div>
          <a className={className} onClick={handleLinkClick} {...others}>
            {children}
          </a>
        </div>
      )
      : (
        <a className={className} {...others}>
          {children}
        </a>
      )}
  </Link>
));


Anchor.defaultProps = {
  noLink: false,
  as: '',
  inheritedStyles: '',
  shallow: false,
  scroll: false,
  forceClick: false,
};

export default styled(Anchor)`${styles}`;

export { Anchor as AnchorVanilla };
