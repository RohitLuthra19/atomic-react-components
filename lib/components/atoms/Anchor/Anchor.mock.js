import { action } from '@storybook/addon-actions';

const defaultAnchor = {
  className: 'example',
  to: 'https://nextjs.org/',
  handleLinkClick: action('clicked'),
};

const secondaryAnchor = {
  className: 'example',
  to: 'https://nextjs.org/',
  handleLinkClick: action('clicked'),
  secondary: true,
};

const openInNewTabAnchor = {
  className: 'example',
  to: 'https://nextjs.org/',
  handleLinkClick: action('clicked'),
  target: '_blank',
};

export { secondaryAnchor, defaultAnchor, openInNewTabAnchor };
