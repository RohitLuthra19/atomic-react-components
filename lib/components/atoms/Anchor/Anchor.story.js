// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import { defaultAnchor, secondaryAnchor, openInNewTabAnchor } from './Anchor.mock';
import Anchor, { AnchorVanilla } from '.';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module).addWithChapters('Anchor', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <AnchorVanilla
              {...defaultAnchor}
              className={`hide-default-sample ${defaultAnchor.className}`}
            >
              Home
            </AnchorVanilla>
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Anchor Variations',
      sections: [
        {
          title: 'Default Link',
          sectionFn: () => <Anchor {...defaultAnchor}>Click Here</Anchor>,
        },
        {
          title: 'Secondary Link',
          sectionFn: () => <Anchor {...secondaryAnchor}>Click Here</Anchor>,
        },
        {
          title: 'Open Link in new Tab',
          sectionFn: () => <Anchor {...openInNewTabAnchor}>Click Here</Anchor>,
        },
      ],
    },
  ],
});
