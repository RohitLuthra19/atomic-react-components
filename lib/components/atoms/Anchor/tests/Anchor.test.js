import React from 'react';
import { shallow, render } from 'enzyme';
import 'jest-styled-components';
import Anchor, { AnchorVanilla } from '../Anchor';

describe('Anchor Component', () => {
  let anchor;

  test('should render correctly and match snapshot', () => {
    anchor = shallow(<Anchor to="/sample">Sample</Anchor>);
    expect(anchor).toMatchSnapshot();
  });

  test('Should handle on typeahead focuson Events', () => {
    const props = { noLink: true, to: '/sample', handleLinkClick: jest.fn() };
    const event = { preventDefault: jest.fn() };
    anchor = shallow(<AnchorVanilla {...props}>Sample</AnchorVanilla>);
    anchor.find('a').simulate('click', event);
    expect(props.handleLinkClick).toHaveBeenCalled();
  });

  test('should check the data attribute of anchor', () => {
    const sampleProps = { elementLocator: 'my-custom-anchor-locator' };
    const anchorVanilla = render(<AnchorVanilla {...sampleProps} />);

    expect(anchorVanilla.attr('data-locator')).toEqual(sampleProps.elementLocator);
  });
});
