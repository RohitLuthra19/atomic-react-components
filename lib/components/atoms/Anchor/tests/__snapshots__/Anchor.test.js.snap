// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed;
}

.c0:focus:not(.focus-visible) {
  outline: none;
}

<Anchor
  as=""
  className="c0"
  forceClick={false}
  handleLinkClick={[Function]}
  inheritedStyles=""
  scroll={false}
  shallow={false}
  to="/sample"
>
  Sample
</Anchor>
`;
